### PR TITLE
Site examples patch version

### PR DESCRIPTION
**Pull request by sdowell**
Issue: GoogleContainerTools/kpt#2609

This adds a new catalog-v2.json endpoint which expands upon the previous catalog.json schema to also include the latest patch version for each major/minor function release. This is presented as a new endpoint in order to avoid breaking compatibility with existing consumers of the original endpoint, such as versions of kpt already in the wild.

Updates the catalog pages of the website to include the latest patch version in:
- The version dropdown for the function
- A new dynamically rendered latest image section

### CODE REVIEW

**Comment by sdowell** [comment]
## Before clicking the dropdown:
image

## After clicking the dropdown
image

**Comment by droot** [comment]
I find that image URL to be really big and distracting. It also doesn't tell what that is. One possible presentation could be having a 

Image: <box> (with the dimensions shown below code section of github)


image

**Comment by droot** [comment]
One more question: if you change the version in the drop down, do the image URL also get changed ? 

I think we can also probably display it under the function heading:

apply-setters
==========

<b>Image URL: </b> `gcr.io/kpt-fn/apply-setters:v0.1.1 `

File: site/index.html [hunk 1]
```diff
@@ -117,9 +117,9 @@
 
               let catalogPath;
               if (isContribFn) {
-                catalogPath = "/contrib/catalog.json";
+                catalogPath = "/contrib/catalog-v2.json";
               } else {
-                catalogPath = "/catalog.json";
```
Comment (mengqiy) [inline site/index.html:122]: Do we still serve catalog.json?

Comment (sdowell) [reply site/index.html:122]: Yes, it's still served at the same path

**Comment by droot** [comment]
> One more question: if you change the version in the drop down, do the image URL also get changed ?
> 
> I think we can also probably display it under the function heading:
> 
> # apply-setters
> Image URL: `gcr.io/kpt-fn/apply-setters:v0.1.1 `

Closing the loop on this. We discussed offline and decided to add this as an incremental change. Scope of this PR is now reduced to just ensuring content in the catalog has latest patch version instead of the minor versions.

**Review by droot — COMMENTED** [review]
Will defer to @mengqiy to approve it.

**Review by mengqiy — APPROVED** [review]

**Pull request merged**